src/main/scala/edu/ie3/simona/model/thermal/CylindricalThermalStorage.scala

```diff
@@ -97,41 +97,49 @@ final case class CylindricalThermalStorage(
   ): (ThermalStorageState, Option[ThermalThreshold]) = {
     /* Determine new state based on time difference and given state */
     val energyBalance = lastState.qDot * Seconds(tick - lastState.tick)
-
     val newEnergy = lastState.storedEnergy + energyBalance
-    val updatedEnergy = newEnergy match {
-      case energy if energy > maxEnergyThreshold => maxEnergyThreshold
-      case energy if energy < minEnergyThreshold => minEnergyThreshold
-      case energy                                => energy
-    }
+    val updatedEnergy =
+      if (isFull(newEnergy))
+        maxEnergyThreshold
+      else if (isEmpty(newEnergy))
+        minEnergyThreshold
```

Comment:
Good so far. However, this could be improved by 
- selecting the correct EnergyThreshold
- changing the flex options accordingly (no more flex in the corresponding direction if EnergyThreshold > (maxEnergyThreshold - toleranceMargin) or if EnergyThreshold < (minEnergyThreshold + toleranceMargin)).

Should be done in another PR